casper/src/test/scala/coop/rchain/casper/MultiParentCasperMergeSpec.scala

```diff
@@ -99,149 +172,236 @@ class MultiParentCasperMergeSpec extends FlatSpec with Matchers with Inspectors
-    "(!C) CX"   -> PERSISTENT_COULD_NOT_MATCH -> merges(S0 | C_, C_, Nil),
-    "(4!) (4!)" -> VOLATILE_EVENT             -> merges(F_ | S0, F_ | S0, Nil),
-    "(4!) (C!)" -> VOLATILE_EVENT             -> merges(F_ | S0, C_ | S0, Nil),
-    "(4!) 4!"   -> VOLATILE_EVENT             -> merges(F1 | S1, F_, S0),
-    "(4!) C!"   -> VOLATILE_EVENT             -> merges(F1 | S1, C_, S0),
-    "(C!) C!"   -> PERSISTENT_COULD_NOT_MATCH -> conflicts(C_ | S0, C_, S0),
-    "4! (4!)"   -> VOLATILE_EVENT             -> merges(F_, F1 | S1, S0),
-    "4! (C!)"   -> PERSISTENT_COULD_NOT_MATCH -> conflicts(F_, C_ | S0, S0),
-    "4X (4!)"   -> VOLATILE_EVENT             -> merges(F_, F_ | S0, Nil),
-    "4X (C!)"   -> PERSISTENT_COULD_NOT_MATCH -> merges(F_, C_ | S0, Nil),
-    "C! (C!)"   -> PERSISTENT_COULD_NOT_MATCH -> merges(C_, C1 | S1, S0),
-    "CX (C!)"   -> PERSISTENT_COULD_NOT_MATCH -> merges(C_, C_ | S0, Nil),
-    "(!4) 4!"   -> VOLATILE_EVENT             -> merges(S1 | F1, F_, S0),
-    "(!4) !C"   -> VOLATILE_EVENT             -> merges(S1 | F1, S0, C0),
-    "(!4) C!"   -> VOLATILE_EVENT             -> merges(S0 | F0, C_, S1),
-    "(4!) CX"   -> VOLATILE_EVENT             -> merges(F_ | S0, C_, Nil),
-    "(C!) (C!)" -> PERSISTENT_COULD_NOT_MATCH -> merges(C_ | S0, C_ | S0, Nil)
+    "!X !X"     -> SamePolarityMerge.desc       -> SamePolarityMerge.mergeable(S0, S0, Nil),
+    "!X !4"     -> SamePolarityMerge.desc       -> SamePolarityMerge.mergeable(S0, S1, F1),
+    "!X (!4)"   -> VolatileEvent.desc           -> VolatileEvent.mergeable(S0)(S0, F_)(Nil),
+    "!X !C"     -> SamePolarityMerge.desc       -> SamePolarityMerge.mergeable(S0, S1, C1),
+    "!X (!C)"   -> PersistentCouldMatch.desc    -> PersistentCouldMatch.conflicting(S0)(S0, C_)(Nil),
+    "!X 4X"     -> IncomingCouldMatch.desc      -> IncomingCouldMatch.conflicting(S0, F_, Nil),
+    "!X 4!"     -> HadItsMatch.desc             -> HadItsMatch.mergeable(S0, F_, S0),
+    "!X (4!)"   -> VolatileEvent.desc           -> coveredBy("!X (!4)"),
+    "!X 4!!"    -> HadItsMatch.desc             -> HadItsMatch.mergeable(S0, F_, R0),
+    "!X (4!!)"  -> PersistentNoMatch.desc       -> PersistentNoMatch.mergeable(S0)(F_, R0)(Nil),
+    "!X !!X"    -> SamePolarityMerge.desc       -> SamePolarityMerge.mergeable(S0, R0, Nil),
+    "!X !!4"    -> SamePolarityMerge.desc       -> SamePolarityMerge.mergeable(S0, R1, F1),
+    "!X (!!4)"  -> PersistentNoMatch.desc       -> coveredBy("!X (4!!)"),
+    "!X CX"     -> IncomingCouldMatch.desc      -> IncomingCouldMatch.conflicting(S0, C_, Nil),
+    "!X C!"     -> IncomingCouldMatch.desc      -> IncomingCouldMatch.conflicting(S0, C_, S0),
+    "!X (C!)"   -> PersistentCouldMatch.desc    -> PersistentCouldMatch.conflicting(S0)(C_, S0)(Nil),
+    "!4 !4"     -> CouldMatchSame.descConflicts -> CouldMatchSame.conflicting(S0, S1, F_),
+    "!4 !4"     -> CouldMatchSame.descMerges    -> CouldMatchSame.mergeable(S0, S1)(F0, F1),
```

Comment:
Notice that in:

`"!4 !4"     -> CouldMatchSame.descMerges    -> CouldMatchSame.mergeable(S0, S1)(F0, F1),`

There's at least 5 kinds of errors possible:
1) mismatch of mergeability category between 1st and 2nd column
2) mismatch of mergeability category between 2nd and 3rd column (due to $FOO.{desc, merges} duplication)
3) mismatch of descMerges/descConflicts with 1st column
4) mismatch of mergable/conflicts with 2nd column
5) mismatch of actual test case with 1st column

The approach I suggest in previous comment elliminates 2) and 4) and will provide way of extension to elliminate the others too.

I'm pointing this out, because I find this crucial: whenever we pass two values separately (vs as a single value) we need to make sure the values are valid w.r.t each other, which requires separate logic outside the place we pass the values in.